src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -306,8 +308,9 @@ public ResponseEntity<Lecture> getLectureWithDetailsAndSlides(@PathVariable Long
     @GetMapping("/lectures/{lectureId}/title")
     @EnforceAtLeastStudent
     public ResponseEntity<String> getLectureTitle(@PathVariable Long lectureId) {
-        final var title = lectureRepository.getLectureTitle(lectureId);
-        return title == null ? ResponseEntity.notFound().build() : ResponseEntity.ok(title);
+        Lecture lecture = lectureRepository.findByIdElseThrow(lectureId);
+        authCheckService.checkIsAllowedToSeeLectureElseThrow(lecture, userRepository.getUserWithGroupsAndAuthorities());
+        return ResponseEntity.ok(lecture.getTitle());
     }
```

Comment:
Can you doublecheck the usage of this endpoint? I think for some reason we decided to not do access checks when retriving only the title.

Reply:
You are right, I reverted the changes I made to that endpoint.